lib/charms/layer/caas_base.py

```diff
@@ -3,26 +3,43 @@
 from pathlib import Path
 from subprocess import run, PIPE, CalledProcessError
 import re
+import tempfile
 
 import yaml
 from charmhelpers.core.hookenv import log
 
 
-def pod_spec_set(spec):
+def pod_spec_set(spec, k8s_resources=None):
```

Comment:
Instead of "k8s_resources", we should keep this name generic, like "custom_resources" or something. Add a function comment might help explain.